prover/circuits/execution/pi.go

```diff
@@ -14,16 +15,16 @@ import (
 // FunctionalPublicInputQSnark the information on this execution that cannot be
 // extracted from other input in the same aggregation batch
 type FunctionalPublicInputQSnark struct {
-	DataChecksum                frontend.Variable
-	L2MessageHashes             L2MessageHashes
-	InitialBlockTimestamp       frontend.Variable
-	FinalStateRootHash          frontend.Variable
-	FinalBlockNumber            frontend.Variable
-	FinalBlockTimestamp         frontend.Variable
-	InitialRollingHashUpdate    [32]frontend.Variable
-	InitialRollingHashMsgNumber frontend.Variable
-	FinalRollingHashUpdate      [32]frontend.Variable
-	FinalRollingHashMsgNumber   frontend.Variable
+	DataChecksum                 frontend.Variable
+	L2MessageHashes              L2MessageHashes
+	InitialBlockTimestamp        frontend.Variable
+	FinalStateRootHash           frontend.Variable
+	FinalBlockNumber             frontend.Variable
+	FinalBlockTimestamp          frontend.Variable
+	InitialRollingHashUpdate     [32]frontend.Variable
+	FirstRollingHashUpdateNumber frontend.Variable
```

Comment:
That's a renaming collectively done with Arya to clarify what the value actually store (the first occurence of a rolling hash update)